Sklep/MainWindow.cs

```diff
@@ -50,42 +50,43 @@ private void BarcodeScanner_CodeScanned(object sender, string code)
             {
                 barCodeTextBox.Text = code;
             }));
-            playCashRegisterBeep();
+            cashRegisterBeep.Play();
             addProductToList(code);
-         
+
         }
         private void addProductToList(string scannedBarcode)
         {
-            if (scannedBarcode != "")
+            if (scannedBarcode == "") return;
+
+            using (var context = new DatabaseContext())
             {
-                var query = db.Products.SingleOrDefault(p => p.Barcode == scannedBarcode);
-                if (query != null)
+                var query = context.Products.SingleOrDefault(p => p.Barcode == scannedBarcode);
+
+                if (query == null) return;
+
+                listOfProducts.Invoke(new MethodInvoker(delegate
                 {
-                    listOfProducts.Invoke(new MethodInvoker(delegate
-                    {
-                        var position = receiptPositionList.Find(p => p.barcode == scannedBarcode);
+                    var position = receiptPositionList.Find(p => p.barcode == scannedBarcode);
 
-                        if (position != null)
-                        {
-                            position.Amount++;
-                        }
-                        else
+                    if (position != null)
+                    {
+                        position.Amount++;
+                    }
+                    else
+                    {
+                        receiptPositionList.Add(new ReceiptPosition()
```

Comment:
tak sobie myślę, że to strasznie dużo kodu jak na dodanie nowej pozycji. Można stworzyć statyczną metodę w `ReceiptPosition.FromProduct(product)`, która zwracałaby `new RecepitPosition()` w oparciu o podany produkt. Albo (o ile to możliwe dla klasy dziedziczącej z Control) dodać nowy konstruktor, który przyjmuje obiekt typu Product.